microsoft/testsuites/osu/osusuite.py

```diff
@@ -0,0 +1,128 @@
+from lisa.tools import Tar, Wget
+
+OSU_MPI_LOCATION = (
+    "https://mvapich.cse.ohio-state.edu/download/mvapich/mv2/mvapich2-2.3.7-1.tar.gz"
+)
+OSU_MPI_FILE_NAME = "mvapich2.tgz"
+
+OSU_BENCH_LOCATION = (
+    "http://mvapich.cse.ohio-state.edu/download/mvapich/osu-micro-benchmarks-6.2.tar.gz"
+)
+OSU_BENCH_FILE_NAME = "osu-bench.tgz"
+
+@TestSuiteMetadata(
+    area="hpc",
+    category="functional",
+    name="OSU Bench",
+    description="""
+    This test suite runs the OSU Micro-Benchmarks MPI test cases.
+    """,
+)
+class OSUTestSuite(TestSuite):
+    TIMEOUT = 200000
+
+    @TestCaseMetadata(
+        description="""
+            This test case runs GPU/CPU MPI latency.
+
+            Steps:
+            1. Install MVAPICH;
+            2. Install OSU Micro_Benchmarks;
+            3. Run GPU/CPU collective/latency tests in a single node.
+
+        """,
+        timeout=TIMEOUT,
+        requirement=simple_requirement(
+            supported_features=[GpuEnabled(), SerialConsole],
+        ),
+        priority=1,
+    )
+    def run_osu_bench(self, node: Node, log: Logger) -> None:
+
+        _install_osu_mpi(node)
+        path = _install_osu_bench(node)
+
+        gpu_count = node.capability.gpu_count
+        gpu = node.features[Gpu]
+        expected_count = gpu.get_gpu_count_with_lspci()
+        assert_that(gpu_count).described_as(
+            "Expected device count didn't match actual device count from lspci!"
+        ).is_equal_to(expected_count)
+
+        node.execute(
+            f"mpirun -np {gpu_count} python3 {path}/python/run.py --benchmark allgather"
+            " --min 128 --max 10000000 >> /tmp/osu_bench.log",  # --buffer pycuda #--buffer cupy
+            shell=True,
+            sudo=True,
+        )
+        log.info("OSU MPI tests finish!")
+
+def _install_osu_mpi(node: Node) -> None:
+
+    wget = node.tools[Wget]
+    tar = node.tools[Tar]
+
+    path = wget.get_tool_path(use_global=True)
+    extracted_path = tar.get_tool_path(use_global=True)
+    if node.shell.exists(path / OSU_MPI_FILE_NAME):
+        return
+
+    download_path = wget.get(
+        url=OSU_MPI_LOCATION, filename=str(OSU_MPI_FILE_NAME), file_path=str(path)
+    )
+    tar.extract(download_path, dest_dir=str(extracted_path))
+
+    target_path = "/usr/local/"
+    node.execute(
+        f"cd {extracted_path}/mvapich2-2.3.7-1; ./configure --prefix {target_path}",
+        shell=True,
+        sudo=False,
+    )
+    node.execute(f"cd {extracted_path}/mvapich2-2.3.7-1; make", shell=True, sudo=False)
+    node.execute(
+        f"cd {extracted_path}/mvapich2-2.3.7-1; make install", shell=True, sudo=True
+    )
+
+    return
+
+def _install_osu_bench(node: Node) -> str:
+
+    wget = node.tools[Wget]
+    tar = node.tools[Tar]
+
+    path = wget.get_tool_path(use_global=True)
+    extracted_path = tar.get_tool_path(use_global=True)
+    if node.shell.exists(path / OSU_BENCH_FILE_NAME):
+        return
```

Comment:
Is this expected return value? The return type of this function is `str`

Reply:
Yes, I was trying to return a file path.

Reply:
The `_install_osu_bench ` is finishing the install and returning the location. The returned location is used in Ln 67. This line is however returning none, which will lead to failure in Ln 67. Can you please instead check if `osu-micro-benchmarks` executable is present and return it's location similar to to Ln 128?